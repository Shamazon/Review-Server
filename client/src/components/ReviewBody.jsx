import style from '../../dist/style/style.css'

const ReviewBody = (props) => {
    return (
        <div className={style.reviewBody}>
            {props.body.split('\n').map((text, index) => React.createElement('p',  { key: index }, text))}
        </div>
    )
}

export default ReviewBody;