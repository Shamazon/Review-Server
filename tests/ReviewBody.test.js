import React from 'react';
import ReviewBody from '../client/src/components/ReviewBody.jsx';
import enzyme from 'enzyme';

describe('ReviewBody', () => {
  it('should be defined', () => {
    expect(ReviewBody).toBeDefined();
  });
  it('should render correctly', () => {
    const props = {
      body: 'test',
    };
    const tree = enzyme.shallow(<ReviewBody {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
