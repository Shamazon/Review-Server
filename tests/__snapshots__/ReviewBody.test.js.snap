// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewBody should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewBody
    body="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          test
        </p>,
      ],
      "className": "reviewBody",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            test
          </p>,
        ],
        "className": "reviewBody",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
